### ISSUE TITLE
Admin panel: Make selecting names easier

### ISSUE THREAD

**Issue by anisfeld** [prijatelilab/PrijateliTree#76]
## What Issue Are You Seeing or What Feature Needs To Be Added?

Once we have 100s of students  in the system, setting up a session will be a pain.
- [x] filter by school name --> for the pilot we'll need a 4th School named "Goce Delcev": https://github.com/prijatelilab/PrijateliTree/pull/81
- [x] type student names / partial completion

**Comment by michplunkett** [comment]
The tables are setup for the logic on filterable tables, it just needs to be programmed in: https://github.com/prijatelilab/PrijateliTree/blob/f9cfb8404dbc2aecd2fa1929b4e7e033be51a62c/prijateli_tree/app/templates/admin_dashboard.html#L7-L12

https://datatables.net/examples/styling/bootstrap5.html

**Comment by anisfeld** [comment]
Advice from ChatGPT:

To implement the functionality of selecting several items from a datatable in JavaScript and then returning those selected items to the database, you can follow these general steps:

1. **Create the DataTable:**
   Use a library like DataTables or any other suitable library to display your data in a table format. Initialize the datatable and make sure it supports row selection.

   Example using DataTables:

   ```html
   <table id="myDataTable">
       <!-- DataTable columns and headers go here -->
   </table>
   ```

   ```javascript
   $(document).ready(function() {
       $('#myDataTable').DataTable({
           // DataTable options and configurations
           select: 'multi', // Enable multi-row selection
       });
   });
   ```

2. **Add Select/Deselect Handlers:**
   Attach event handlers to capture selected rows. DataTables provides events like `select` and `deselect` which you can use.

   ```javascript
   $('#myDataTable').on('select.dt deselect.dt', function() {
       // Handle selection/deselection logic here
       var selectedRows = $('#myDataTable').DataTable().rows({ selected: true }).data().toArray();
       console.log(selectedRows);
   });
   ```

3. **Store Selected Items:**
   Maintain a data structure (e.g., an array) to store the selected items in memory.

   ```javascript
   var selectedItems = [];

   $('#myDataTable').on('select.dt deselect.dt', function() {
       selectedItems = $('#myDataTable').DataTable().rows({ selected: true }).data().toArray();
   });
   ```

4. **Send Selected Items to the Database:**
   When the user is done with the selection, send the selected items to the server-side for updating the database.

   ```javascript
   function saveSelectedItems() {
       // Assuming you are using AJAX to send data to the server
       $.ajax({
           url: '/api/updateDatabase',
           method: 'POST',
           contentType: 'application/json',
           data: JSON.stringify(selectedItems),
           success: function(response) {
               console.log('Data updated successfully:', response);
           },
           error: function(error) {
               console.error('Error updating data:', error);
           }
       });
   }
   ```

5. **Trigger the Database Update:**
   You can trigger the database update when the user clicks a button or performs any action to confirm their selection.

   ```html
   <button onclick="saveSelectedItems()">Save Selection</button>
   ```

   You can, of course, customize and adapt these steps based on your specific use case and application architecture. Additionally, ensure that you handle security aspects such as input validation and protection against SQL injection when interacting with the database.

**Comment by anisfeld** [comment]
Now we have a DataTable that allows the administrator 
- to filter using the search bar
- to select participants by clicking on the names of  players
- to get client-side feedback on whether the selection works (6 players, 3 from a given language group)
 

![image](https://github.com/prijatelilab/PrijateliTree/assets/25271399/4f3cbc61-6a51-4ab6-9158-fd29e39c6b26)

Remaining potential issues:
- The Admin cannot easily see which rows are selected, seems like an easy source of annoyance if they make a mistake once there are 100s of students in the study.
- The Admin can filter the data by high school or grade using the search tool, but this seems less intuitive.
- Admin cannot select the number of games in this setting (more of an issue during piloting, where we are still playing with this).

**Comment by anisfeld** [comment]
closed with #140
